tests/behat/features/gridfield-search.feature

```diff
@@ -23,3 +28,22 @@ Feature: Search in GridField
       And I click "Walmart" in the "#Form_EditForm" element
     Then I should see "Walmart"
       And I should see "Walmart" in the ".breadcrumbs-wrapper" element
+
+  Scenario: I can clear search and paginate
+    When I press the "Open search and filter" button
+    # This will match on all of the "Walmart" records
+    And I fill in "SearchBox__q" with "a a"
+    And I press the "Enter" key in the "SearchBox__q" field
+    Then the "[name=SearchBox__q]" element "value" attribute should be "a a"
+    And I should see "Walmart" in the ".ss-gridfield-item:nth-of-type(1) .col-Name" element
+    And I should see "Walmart B" in the ".ss-gridfield-item:nth-of-type(2) .col-Name" element
+    And I should see "Walmart C" in the ".ss-gridfield-item:nth-of-type(3) .col-Name" element
+    And I should not see "ExxonMobil" in the ".col-Name" element
+    And I should not see "Vitol" in the ".col-Name" element
+    And I click on the ".search-box__cancel" element
+    When I click on the "#action_pagination_next" element
+    And I press the "Open search and filter" button
+    Then the "[name=SearchBox__q]" element "value" attribute should be ""
+    And I should see "Walmart B" in the ".ss-gridfield-item:nth-of-type(1) .col-Name" element
+    And I should see "Walmart C" in the ".ss-gridfield-item:nth-of-type(2) .col-Name" element
+    And I should see "Walmart D" in the ".ss-gridfield-item:nth-of-type(3) .col-Name" element
```

Comment:
Sorry, could we check that 'ExxonMobil' also exist in the list.

Reply:
It doesn't show because there's a pagination limit of 3 due to the use of the `FrameworkTestModelAdminExtension` and we're paginated to the 2nd page after clicking the `#action_pagination_next` element